wolfcrypt/src/random.c

```diff
@@ -3415,6 +3415,70 @@ int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
         return 0;
     }
 
+#elif defined(ARDUINO)
+
+    int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
+    {
+        int ret = 0;
+        word32 rand;
+        while (sz > 0) {
+            word32 len = sizeof(rand);
+            if (sz < len)
+                len = sz;
+        /* Get an Arduino framework random number */
+        #if defined(__arm__)
+            /* See: https://github.com/avrxml/asf/tree/master/sam/utils/cmsis/sam3x/include */
+            #if defined(__SAM3A4C__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3A8C__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3X4C__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3X4E__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3X8C__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif defined(__SAM3X8E__)
+                /* This is the Arduino Due */
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #elif  defined(__SAM3A8H__)
+                #ifndef TRNG
+                    #define TRNG (0x400BC000U)
+                #endif
+            #else
+                #ifndef TRNG
+                    #error "Unknown TRNG for this device"
+                #endif
+            #endif
+
+            srand(analogRead(0));
+            rand = trng_read_output_data(TRNG);
+        #elif defined(__STM32__)
+            /* TODO: confirm this is proper random number on Arduino STM32 */
+            rand = random();
+        #else
+            rand = random();
```

Comment:
Consider `#warning` or `pragma warn`

Reply:
Agreed. I've added the warning. 

Note the Arduino IDE by default suppresses all compiler messages. See `File` - `Preferences` - `Compiler Warnings`

Reply:
Still no warning here for this case.

Reply:
There _is_ a warning located on [line 3470 of my fork for this PR](https://github.com/gojimmypi/wolfssl/blob/105c4cb47bd3ba6863384fc636f0c40595ce25d3/wolfcrypt/src/random.c#L3470).

I cannot explain why the GitHub preview, above, clearly shows _no warning_ between the comment and the code, and yet when I click on the link to `wolfcrypt/src/random.c` and scroll down the the `random.c` file, the warning is there:


![image](https://github.com/wolfSSL/wolfssl/assets/13059545/ca4a3f15-eb10-4e06-961d-0ac82f47e2c8)

Reply:
That warning will not be hit for the `#else` case here at line 3473.

Reply:
That was intended, as I had assumed that other cases would be able to use the native Arduino implementation of `random()`. 

Upon reconsideration I agree it is best to only have that for known, tested targets and warn for all others, including the `else`. I've added the warning.